internal/mithril-persistence/src/sqlite/transaction.rs

```diff
@@ -0,0 +1,174 @@
+use crate::sqlite::SqliteConnection;
+
+/// Sqlite transaction wrapper.
+///
+/// Transactions are automatically rolled back if this struct object is dropped and
+/// the transaction was not committed.
+pub struct Transaction<'a> {
+    connection: &'a SqliteConnection,
+    // An active transaction is one that has yet to be committed or rolled back.
+    is_active: bool,
+}
+
+impl<'a> Transaction<'a> {
+    /// Begin a new transaction.
+    pub fn begin(connection: &'a SqliteConnection) -> Result<Self, sqlite::Error> {
+        connection.execute("BEGIN TRANSACTION")?;
+        Ok(Self {
+            connection,
+            is_active: true,
+        })
+    }
+
+    /// Commit the transaction.
+    pub fn commit(mut self) -> Result<(), sqlite::Error> {
+        self.is_active = false;
+        self.connection.execute("COMMIT TRANSACTION")?;
+        Ok(())
+    }
+
+    /// Rollback the transaction.
+    pub fn rollback(mut self) -> Result<(), sqlite::Error> {
+        self.is_active = false;
+        self.connection.execute("ROLLBACK TRANSACTION")?;
+        Ok(())
+    }
+}
+
+impl Drop for Transaction<'_> {
+    fn drop(&mut self) {
+        if self.is_active {
+            // Unwrap should not happen here, otherwise it would mean that we have not handled
+            // correctly the transaction "active" state or that the connection was closed.
+            self.connection.execute("ROLLBACK TRANSACTION").unwrap();
```

Comment:
We could call rollback function here.
Even if it's not necessary to set is_active to false in the drop function, it may help to understand states

Reply:
Sadly we can't call `rollback` here since it move the struct and `drop` only take the struct as a borrowed mut.

Instead I defined a constant for the `ROLLBACK TRANSACTION` instruction.